packages/core/src/api/blockManipulation/commands/getBlock/getBlock.ts

```diff
@@ -0,0 +1,127 @@
+import { Node } from "prosemirror-model";
+
+import { Block } from "../../../../blocks/defaultBlocks.js";
+import type { BlockNoteEditor } from "../../../../editor/BlockNoteEditor.js";
+import {
+  BlockIdentifier,
+  BlockSchema,
+  InlineContentSchema,
+  StyleSchema,
+} from "../../../../schema/index.js";
+import { nodeToBlock } from "../../../nodeConversions/nodeToBlock.js";
+import { getNodeById } from "../../../nodeUtil.js";
+
+// Get a block from a block identifier, or return undefined if the block cannot
+// be found in the editor. The ProseMirror node that gets converted into a block
+// is the one that has the same ID as the block identifier. However, if the
+// `getDifferentNodeForConversion` function is provided, the node returned from
+// it will be converted instead, e.g. a sibling or parent. If
+// `getDifferentNodeForConversion` returns undefined, this function will also
+// return undefined.
+function getBlockFromNode<
+  BSchema extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(
+  editor: BlockNoteEditor<BSchema, I, S>,
+  blockIdentifier: BlockIdentifier,
+  getDifferentNodeForConversion: ({
+    node,
+    posBeforeNode,
+  }: {
+    node: Node;
+    posBeforeNode: number;
+  }) => Node | undefined = ({ node }) => node
+): Block<BSchema, I, S> | undefined {
+  const id =
+    typeof blockIdentifier === "string" ? blockIdentifier : blockIdentifier.id;
+
+  try {
+    const nodeToConvert = getDifferentNodeForConversion(
+      getNodeById(id, editor._tiptapEditor.state.doc)
+    );
+    if (!nodeToConvert) {
+      return undefined;
+    }
+
+    return nodeToBlock(
+      nodeToConvert,
+      editor.schema.blockSchema,
+      editor.schema.inlineContentSchema,
+      editor.schema.styleSchema,
+      editor.blockCache
+    );
+  } catch (e) {
```

Comment:
why is there a try / catch? Seems suspicious, can we remove it?

Reply:
`getNodeById` throws an error if a node with the given ID cannot be found, but for `getBlocks`, we just want to return undefined if the block can't be found. Maybe we want to also return undefined instead of throwing an error for `getNodeById`? Not sure